dali/operators/reader/nvdecoder/sequencewrapper.h

```diff
@@ -47,44 +47,28 @@ struct SequenceWrapper {
     timestamps.clear();
     timestamps.reserve(count);
 
-    int dev;
-    CUDA_CALL(cudaGetDevice(&dev));
-    std::unique_lock<std::mutex> lock{started_lock_};
-    if (started_) {
-      CUDA_CALL(cudaEventDestroy(event_));
+    if (!initialized_) {
+      CUDA_CALL(cudaEventCreateWithFlags(&event_, cudaEventBlockingSync | cudaEventDisableTiming));
```

Comment:
Why not use CUDAEvent instead? It has a proper destructor - no need to write your own.

Reply:
Legacy..., I have also forgot about our wrappers, fixed